sympy/utilities/tests/test_lambdify.py

```diff
@@ -1340,6 +1341,24 @@ def test_issue_22739():
     assert abs(f.subs(point) - F(*point.values())) <= 1e-10
 
 
+def test_issue_22992():
+    if not numpy:
+        skip("numpy not installed")
+
+    a, t = symbols('a t')
+    expr = a*(log(cot(t/2)) - cos(t))
+    F = lambdify([a, t], expr, 'numpy')
+
+    point = {a: 10, t: 2}
+
+    assert abs(expr.subs(point) - F(*point.values())) <= 1e-10
+
+    # Standard math
+    F = lambdify([a, t], expr)
+
+    assert abs(expr.subs(point) - F(*point.values())) <= 1e-10
```

Comment:
Are there any choices of `point` that would deviate substantially between `cot` and `tan**-1` due to numerical implementation? I'd argue that this test would imply to users that this accuracy holds for any point, but that might not actually be the case.

Reply:
I wouldn't have thought that there would be a substantial deviation. Rounding error from division is generally minimal.

For inverse functions there could be issues with branch cuts and signed zeros (https://github.com/sympy/sympy/issues/21751)

Reply:
I guess one can probably find cases where it is better to evaluate the original function as opposed to a rewrite, but to some extent it is choice where to get the errors... And in most cases, I assume that the rewrites are OK enough and not worse that just evaluating a large symbolic argument to `cot` like `sqrt(sqrt(2) + sin(pi/7))` or so.